core/storage/btree.rs

```diff
@@ -145,57 +149,114 @@ impl BTreeCursor {
+        match self.move_to(key.clone(), op.clone())? {
+            CursorResult::Ok(_) => {}
+            CursorResult::IO => return Ok(CursorResult::IO),
+        };
 
         let mem_page = self.get_mem_page();
-
         let page_idx = mem_page.page_idx;
         let page = self.pager.read_page(page_idx)?;
         let page = RefCell::borrow(&page);
         if page.is_locked() {
             return Ok(CursorResult::IO);
         }
+
         let page = page.contents.read().unwrap();
         let page = page.as_ref().unwrap();
 
         for cell_idx in 0..page.cell_count() {
-            match &page.cell_get(
+            let cell = page.cell_get(
                 cell_idx,
                 self.pager.clone(),
                 self.max_local(page.page_type()),
                 self.min_local(page.page_type()),
                 self.usable_space(),
-            )? {
+            )?;
+            match &cell {
                 BTreeCell::TableLeafCell(TableLeafCell {
                     _rowid: cell_rowid,
-                    _payload: p,
+                    _payload: payload,
                     first_overflow_page: _,
                 }) => {
-                    if *cell_rowid == rowid {
-                        let record = crate::storage::sqlite3_ondisk::read_record(p)?;
+                    let SeekKey::TableRowId(rowid_key) = key else {
+                        unreachable!("table seek key should be a rowid");
+                    };
+                    mem_page.advance();
+                    let comparison = match op {
+                        SeekOp::GT => *cell_rowid > rowid_key,
+                        SeekOp::GE => *cell_rowid >= rowid_key,
+                        SeekOp::EQ => *cell_rowid == rowid_key,
+                    };
+                    if comparison {
+                        let record = crate::storage::sqlite3_ondisk::read_record(payload)?;
                         return Ok(CursorResult::Ok((Some(*cell_rowid), Some(record))));
                     }
                 }
+                BTreeCell::IndexLeafCell(IndexLeafCell { payload, .. }) => {
+                    let SeekKey::IndexKey(index_key) = key else {
+                        unreachable!("index seek key should be a record");
+                    };
+                    mem_page.advance();
+                    let record = crate::storage::sqlite3_ondisk::read_record(payload)?;
+                    let comparison = match op {
+                        SeekOp::GT => record > *index_key,
+                        SeekOp::GE => record >= *index_key,
+                        SeekOp::EQ => record == *index_key,
+                    };
+                    if comparison {
+                        let rowid = match record.values.get(1) {
+                            Some(OwnedValue::Integer(rowid)) => *rowid as u64,
+                            _ => unreachable!("index cells should have an integer rowid"),
+                        };
```

Comment:
I don't get this part. We assume `record.values.get(1)` is the rowid. But I imagine there could be indexes with multiple columns right? Shouldn't the rowid be the last value? Furthermore, should we maybe change rowid name to key or something similar because it can be eitther rowid and primary key I think.

Reply:
Thats a bug i forgot to refactor. it should be `last`

Reply:
Fixed in https://github.com/penberg/limbo/pull/350/commits/93a8110773e764c8507a2b4cd64187567c7b78f6